platform/view/core/endpoint/endpoint.go

```diff
@@ -241,20 +237,20 @@ func (r *Service) SetPublicKeyIDSynthesizer(publicKeyIDSynthesizer driver.Public
 }
 
 func (r *Service) pkiResolve(resolver *Resolver) []byte {
-	r.pkiExtractorsLock.RLock()
+	resolver.PKILock.Lock()
```

Comment:
can we have a read lock here and if len(resolver.PKI) == 0 then we get a write lock?